pkg/provider/ibmcloud/plugin/loadbalancer/lb.go

```diff
@@ -0,0 +1,466 @@
+		return lbResult(""), fmt.Errorf("unknown port %v", extPort)
+	}
+
+	return lbResult("unpublish"), nil
+}
+
+// ConfigureHealthCheck configures the health checks for instance removal and reconfiguration
+// The parameters healthy and unhealthy indicate the number of consecutive success or fail pings required to
+// mark a backend instance as healthy or unhealthy. The ping occurs on the backendPort parameter and
+// at the interval specified.
+func (l *ibmcloudlb) ConfigureHealthCheck(hc loadbalancer.HealthCheck) (loadbalancer.Result, error) {
+	lbLogger.Debug("ConfigureHealthCheck", "name", l.name, "heathCheck", hc, "V", debugV)
+	l.lock.Lock()
+	defer l.lock.Unlock()
+
+	hmConfig := datatypes.Network_LBaaS_LoadBalancerHealthMonitorConfiguration{
+		BackendPort: &hc.BackendPort,
+		// IBM Clouud only has a single retry config, so make it the max of the two: healthy and unhealthy
+		MaxRetries: intPtr(max(hc.Healthy, hc.Unhealthy)),
+		Interval:   intPtr(int(hc.Interval / time.Second)),
+		Timeout:    intPtr(int(hc.Timeout / time.Second)),
+	}
+	hmArray := []datatypes.Network_LBaaS_LoadBalancerHealthMonitorConfiguration{hmConfig}
+
+	_, err := l.lbHealthMonitorService.UpdateLoadBalancerHealthMonitors(&l.uuid, hmArray)
+	if err != nil {
+		return nil, err
+	}
+
+	return lbResult("healthcheck"), nil
+}
+
+// RegisterBackend registers instances identified by the IDs to the LB's backend pool
+func (l *ibmcloudlb) RegisterBackends(ids []instance.ID) (loadbalancer.Result, error) {
+	lbLogger.Debug("RegisterBackends", "name", l.name, "ids", ids, "V", debugV)
+	l.lock.Lock()
+	defer l.lock.Unlock()
+
+	// Set default weight to 50 to match LB default
+	weight := 50
+
+	// Output array of members added
+	membersToAdd := []datatypes.Network_LBaaS_LoadBalancerServerInstanceInfo{}
+
+	// Get current list of backend members
+	lb, err := l.lbService.Mask("members").GetLoadBalancer(&l.uuid)
+	if err != nil {
+		return nil, err
+	}
+
+	// Iterate over the list of members to add checking if they already exist in the loadbalancer.
+	// If not, add member.
+	for _, id := range ids {
+		ip := string(id)
+		ipFound := false
+		for _, member := range lb.Members {
+			if *member.Address == ip {
+				ipFound = true
+				lbLogger.Error("RegisterBackends", "name", l.name, "V", debugV, "backend already exists", ip)
+				break
+			}
+		}
+		if ipFound == false {
+			member := datatypes.Network_LBaaS_LoadBalancerServerInstanceInfo{
+				PrivateIpAddress: &ip,
+				Weight:           &weight,
+			}
+			membersToAdd = append(membersToAdd, member)
+		}
+	}
+
+	// If there are any members to add, add them.
+	if len(membersToAdd) > 0 {
+		attempt := 0
+		for {
+			attempt++
+			_, err = l.lbMemberService.AddLoadBalancerMembers(&l.uuid, membersToAdd)
+			if err == nil {
+				break
+			}
+			// Check for the lb is in state UPDATE_PENDING. (HTTP 500)
+			if strings.Contains(err.Error(), "UPDATE_PENDING") && attempt <= updateRetryCount {
+				time.Sleep(updateRetryTime)
```

Comment:
Can we add in some logging here that exposes this error case? I like that we are handling it but it would be nice to get visibility into how often this happens.

Reply:
Done in the several places we have this code.